runtime/doc/lua.txt

```diff
@@ -2586,6 +2586,33 @@ vim.uri_to_fname({uri})                                   *vim.uri_to_fname()*
 ==============================================================================
 Lua module: vim.ui                                                    *vim.ui*
 
+*vim.ui.Spinner*
+
+    Fields: ~
+      • {steps}    (`vim.ui.SpinnerStep[]`) Array of spinner steps. See
+                   |vim.ui.SpinnerStep|.
+      • {on_step}  (`fun(char: string, char_id: integer)`) Callback to be
+                   executed.
+      • {reverse}  (`boolean`) Whether to reverse from last to first before
+                   repeating a cycle.
+      • {start}    (`fun(self: vim.ui.Spinner)`) See |Spinner:start()|.
+      • {stop}     (`fun(self: vim.ui.Spinner)`) See |Spinner:stop()|.
+
+*vim.ui.SpinnerStep*
+    before showing next step).
+
+    Fields: ~
+      • {char}   (`string`) Character to be displayed as snipper step.
```

Comment:
After sleeping on this, I don't think `char` is the best name here. In theory, there is nothing wrong with showing spinner with more than a single character. For example, like [this one](https://github.com/sindresorhus/cli-spinners/blob/a01e4cf1235e1425f0243f36887db168cf710279/spinners.json#L656).

Not sure about the appropriate name here, though. Can be `str` for "string". Can also be `frame`, as used in 'sindresorhus/cli-spinners'.

Reply:
prefer `text` instead of `str`.